lib/settings/pages/general_settings_page.dart

```diff
@@ -221,13 +236,24 @@ class _GeneralSettingsPageState extends State<GeneralSettingsPage> with SingleTi
       communitySeparator = FullNameSeparator.values.byName(prefs.getString(LocalSettings.communityFormat.name) ?? FullNameSeparator.dot.name);
 
       showInAppUpdateNotification = prefs.getBool(LocalSettings.showInAppUpdateNotification.name) ?? false;
+      enableInboxNotifications = prefs.getBool(LocalSettings.enableInboxNotifications.name) ?? false;
     });
   }
 
   @override
   void initState() {
     super.initState();
-    WidgetsBinding.instance.addPostFrameCallback((_) => _initPreferences());
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      _initPreferences();
+
+      // Check whether Android is currently allowing Thunder to send notifications
```

Comment:
I would place all this logic into its own function (e.g., similar to `_initPreferences`). This way, you can perform the async operations!

Reply:
Ahh, I think I misinterpreted the comment. We can do something like this so that it doesn't mutate `areAndroidNotificationsAllowed` directly:

Although in this case, it might be fine!

```dart
bool _areAndroidNotificationsAllowed = await androidFlutterLocalNotificationsPlugin?.areNotificationsEnabled();

setState(() {
  areAndroidNotificationsAllowed: _areAndroidNotificationsAllowed,
});
```

Reply:
Ok I updated it to do this! I think the end result is the same (the UI is redrawn with the current state), but I agree that this provides a clearer intention as to why `setState` is being called at all.